cfgmgr/buffermgrdyn.cpp

```diff
@@ -268,11 +270,82 @@ string BufferMgrDynamic::getDynamicProfileName(const string &speed, const string
     return buffer_profile_key + "_profile";
 }
 
+string BufferMgrDynamic::getMaxSpeedFromList(string speedList)
+{
+    auto &&speedVec = tokenize(speedList, ',');
+    unsigned long maxSpeedNum = 0, speedNum;
+    string maxSpeedStr;
+    for (auto &speedStr : speedVec)
+    {
+        speedNum = atol(speedStr.c_str());
+        if (speedNum > maxSpeedNum)
+        {
+            maxSpeedNum = speedNum;
+            maxSpeedStr = speedStr;
```

Comment:
would be better to get rid of this var and convert to string just before returning

Reply:
Updated.